doc/voq/aggregate_voq_counters.md

```diff
@@ -0,0 +1,128 @@
+}
+```
+
+The VOQ stats will be updated in a new table `COUNTERS_VOQ`
+
+The following new VOQ counters should be available for each VOQ entry in the DB:
+   * `COUNTERS_VOQ : LINECARD | ASIC | EthernetXXX @ LINECARD | ASIC : VOQ_index`
+     * `SAI_VOQ_STAT_PACKETS`
+     * `SAI_VOQ_STAT_BYTES`
+     * `SAI_VOQ_STAT_DROPPED_PACKETS`
+     * `SAI_VOQ_STAT_DROPPED_BYTES`
+     * `SAI_QUEUE_STAT_CREDIT_WD_DELETED_PACKETS`
+
+   * `COUNTERS_VOQ` is the table name.
+   * The first part of the key ( before `@` ) `LINECARD | ASIC | EthernetXXX` denotes the physical location of the interface ( or full system port name )
+   * The second part of the key ( after `@` ) `LINECARD | ASIC` denotes the location of the VOQ.
+   * VOQ_index is the index of the VOQ in question.
+        
+#### SWSS Changes
+##### New VoqStatsOrch module
+A new module called VoqStatsOrch will be introduced which will be initialised by orchdaemon.
+
+VoqStatsOrch will synchronise the VOQ counters between each ASIC's COUNTERS_DB on linecards and CHASSIS_COUNTERS_DB running on the supervisor.
+
+#### gNMI changes
+New virtual paths will be introduced to retrieve VOQ counters from linecard and aggregated VOQ counter stats from supervisor
+
+|  DB target|   Virtual Path  | Supported On? |     Description|
+|  ----     |:----:| :-:| ----|
+|COUNTERS_DB | "COUNTERS/``<asic id>``/``<system port>``/Voq"| Linecard |  All VOQ counters for a sytem port on an ASIC on linecard
+|COUNTERS_DB | "COUNTERS/``<asic id>``/``*``/Voq"| Linecard | All VOQ counters for all sytem ports on an ASIC on linecard
+|COUNTERS_DB | "COUNTERS/``<system port>``/Voq"| Supervisor | Aggregated VOQ counters for a system port from supervisor
+|COUNTERS_DB | "COUNTERS/``*``/Voq"| Supervisor | Aggregated VOQ counters for all system ports from supervisor
+
+Note: For the sake of uniformity the virtual path for supervisor says target as `COUNTERS_DB` and table as `COUNTERS` but it will be internally mapped to `CHASSIS_COUNTERS_DB` and `COUNTERS_VOQ`.
+
+#### Repositories that need to be changed
+   * sonic-buildimage 
+   * sonic-swss-common 
+   * sonic-swss 
+   * sonic-utilities 
+   * sonic-gnmi 
+   * sonic-mgmt 
+
+### SAI API 
+No new SAI API is being added. 
+
+### Configuration and management 
+#### CLI
+CLI (queuestat.py) aggregates the VOQ stats for a VOQ across ASICS and present a consolidated view. No new CLI command is being introduced for this rather the following CLI command is leveraged to provide this output on an SSI.
+
+$ show VOQ counters [interface] --voq
+```
+admin@cmp217:~$ show queue counters "cmp217-5|asic0|Ethernet24" --voq
+                     Port    Voq    Counter/pkts    Counter/bytes    Drop/pkts    Drop/bytes    Credit-WD-Del/pkts
+-------------------------  -----  --------------  ---------------  -----------  ------------  --------------------
+cmp217-5|asic0|Ethernet24   VOQ0              54             2700            0             0                     0
```

Comment:
What are the other filters we can apply for this show command?. Can we filter the output for a Line card or asic in Linecard?

Reply:
No other filter can be applied as of now. We can only aggregated counters as per system port.